File: cpp_src/ImplementStrStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int haystackSize = haystack.size();
        int needleSize = needle.size();
        if(needleSize == 0){
            return 0;
        }
        int last = haystackSize - needleSize;
        int i,j;
        for(i = 0; i <= last; i++){
            for(j = 0; j < needleSize; j++){
                if(haystack[i + j] != needle[j]) break;
            }
            if(j == needleSize){
                return i;
            }
        }
        return -1;
    }
};

File: python_src/ImplementStrStr.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if len(haystack) == 0:
            if len(needle) == 0: return 0
            else: return -1
        for i in range(len(haystack) - len(needle) + 1):
            for j in range(len(needle)):
                if haystack[i+j] != needle[j]: break
            else: return i
        return -1
        